internal/channel/channel.go

```diff
@@ -20,109 +18,153 @@ type Channel struct {
-	}
+// initPlugin returns a new Plugin or nil if an error occurred during initialization
+func (c *Channel) initPlugin() *Plugin {
+	c.Logger.Debug("Initializing channel plugin")
 
-	reader, err := cmd.StdoutPipe()
+	p, err := NewPlugin(c.Type, c.Logger)
 	if err != nil {
-		return fmt.Errorf("failed to create stdout pipe: %w", err)
+		c.Logger.Errorw("Failed to initialize channel plugin", zap.Error(err))
+		return nil
 	}
 
-	errPipe, err := cmd.StderrPipe()
-	if err != nil {
-		return fmt.Errorf("failed to create stderr pipe: %w", err)
+	if err := p.SetConfig(c.Config); err != nil {
+		c.Logger.Errorw("Failed to set channel plugin config, terminating the plugin", zap.Error(err))
+		p.Stop()
+		return nil
 	}
 
-	go forwardLogs(errPipe, c.Logger)
-
-	if err = cmd.Start(); err != nil {
-		return fmt.Errorf("failed to start cmd: %w", err)
-	}
-	c.Logger.Debug("Cmd started successfully")
+	return p
+}
 
-	c.cmd = cmd
-	c.rpc = rpc.NewRPC(writer, reader, c.Logger)
+// runPlugin is called as go routine to initialize and maintain the plugin by receiving signals on given chan(s)
+func (c *Channel) runPlugin() {
+	var currentlyRunningPlugin *Plugin
+	for {
+		if currentlyRunningPlugin == nil {
+			currentlyRunningPlugin = c.initPlugin()
+		}
 
-	if err = c.SetConfig(c.Config); err != nil {
-		go c.terminate(c.cmd, c.rpc)
+		var rpcDone <-chan struct{}
+		if currentlyRunningPlugin != nil {
+			rpcDone = currentlyRunningPlugin.rpc.Done()
+		}
 
-		c.cmd = nil
-		c.rpc = nil
+		select {
+		case <-rpcDone:
+			c.Logger.Debug("Receive plugin error signal")
 
-		return fmt.Errorf("failed to set config: %w", err)
-	}
-	c.Logger.Debugw("Successfully set config", zap.String("config", c.Config))
+			if currentlyRunningPlugin != nil {
+				currentlyRunningPlugin.Stop()
+				currentlyRunningPlugin = nil
+			}
 
-	return nil
-}
+			continue
+		case <-c.newConfigCh:
+			c.Logger.Debug("Receive reload plugin config signal")
 
-func (c *Channel) Stop() {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+			if currentlyRunningPlugin != nil {
+				currentlyRunningPlugin.Stop()
+				currentlyRunningPlugin = nil
+			}
 
-	if c.cmd == nil {
-		c.Logger.Debug("channel plugin has already been stopped")
-		return
-	}
+			continue
+		case <-c.stopPluginCh:
+			c.Logger.Debug("Receive plugin stop signal")
 
-	go c.terminate(c.cmd, c.rpc)
+			if currentlyRunningPlugin != nil {
+				currentlyRunningPlugin.Stop()
+				currentlyRunningPlugin = nil
+			}
 
-	c.cmd = nil
-	c.rpc = nil
+			return
+		case req := <-c.notificationCh:
+			if currentlyRunningPlugin == nil {
+				currentlyRunningPlugin = c.initPlugin()
+			}
 
-	c.Logger.Debug("Stopped channel plugin successfully")
+			if currentlyRunningPlugin == nil {
+				c.Logger.Debug("Cannot send notification, plugin could not be started")
+				req.errCh <- errors.New("plugin could not be started")
+			} else {
+				go func(p *Plugin) {
+					// the return can take time
+					req.errCh <- p.SendNotification(req.req)
+				}(currentlyRunningPlugin)
```

Comment:
After looking on this piece of code for a bit, we can also do this slightly different:
1. Remove `Req`, `notificationCh` and this case.
2. Add a `chan *Plugin` to the `Channel` struct.
3. Add a `case c.pluginCh <- currentlyRunningPlugin:` here (with no code in that case).

Now this gives a channel that you can receive from to get a `*Plugin` that can be used to send a notification. Slight caveat: the first `if currentlyRunningPlugin == nil` that attempts to start the plugin once again after a long wait is lost. But that can be addressed easily with a simple function:

```go
// getPlugin returns a fully initialized plugin that can be used for sending notifications. If there
// currently is no such plugin, for example because starting it failed, nil is returned instead.
func (c *Channel) getPlugin() *Plugin {
    p := <-c.pluginCh
    
    if p == nil {
        // The above receive might have woken runPlugin after the select was blocked for a long time.
        // In that case, a second receive gives it another chance to successfully start the plugin.
        p = <-c.pluginCh
    }

    return p
}
```

And then, `Notify()` could just do the following:

```go
p := c.getPlugin()
if p == nil {
    // TODO: error handling
}
p.SendNotification(req.req)
```

Overall, has the same effect as the current code: the value of `currentlyRunningPlugin` is passed to a different goroutine which calls `SendNotification()` on it. But this no longer needs the `Req` type, no passing around of channels for return values, and no extra goroutines need to be started.